lib/awful/keygrabber.lua

```diff
@@ -480,7 +480,7 @@ end
 -- @method stop
 -- @emits stopped
 -- @emits property::current_instance
-function keygrabber:stop(_stop_key, _stop_mods) -- (at)function disables ldoc params
+function keygrabber:stop(_stop_key, _stop_mods) --luacheck: ignore 214
```

Comment:
What's the reason for ignoring these? There'd be no shadowing as far as I can tell.

Reply:
It's because they are undocumented. We overload the meaning of `_` a bit between "it's private" and "it's undocumented, thus private". Luacheck folks randomly decided to implement a naming convention.

Reply:
A public function with private parameters? And overloading a naming convention with a completely orthogonal meaning?

I think the much saner solution would be to document those parameters and talk about why the end user shouldn't use them, than to jump through multiple confusing hoops to hide the documentation for something the user can easily access.
Clear documentation will also help users that get confusing error messages when they accidentally pass a value, pointing to a `_stop_key` parameter in a function that, as per documentation, shouldn't have parameters.

Reply:
Fair point, done.